src/cargo/util/toml_mut/manifest.rs

```diff
@@ -276,21 +276,24 @@ impl LocalManifest {
 
     /// Write changes back to the file.
     pub fn write(&self) -> CargoResult<()> {
+        // cargo update --breaking needs to be able to write a workspace root manifest.
+
         if !self.manifest.data.contains_key("package")
             && !self.manifest.data.contains_key("project")
+            && !self.manifest.data.contains_key("workspace")
         {
-            if self.manifest.data.contains_key("workspace") {
-                anyhow::bail!(
-                    "found virtual manifest at {}, but this command requires running against an \
-                         actual package in this workspace.",
-                    self.path.display()
-                );
-            } else {
-                anyhow::bail!(
-                    "missing expected `package` or `project` fields in {}",
-                    self.path.display()
-                );
-            }
+            // if self.manifest.data.contains_key("workspace") {
+            //     anyhow::bail!(
+            //         "found virtual manifest at {}, but this command requires running against an \
+            //              actual package in this workspace.",
+            //         self.path.display()
+            //     );
+            // } else {
+            anyhow::bail!(
+                "missing expected `package`, `project`, or `workspace` fields in {}",
+                self.path.display()
+            );
+            // }
```

Comment:
To be resolved

Reply:
Does the change look OK to you?

Reply:
We should just remove the error check.  I think its an artifact of a previous version of `cargo add` from before it got merged into cargo and it had to do more of the heavy lifting.

Reply:
Removed.